server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -458,6 +458,26 @@ private void write(final Collection<CommitSession> sessions, boolean mincFinish,
       // the logs haven't changed.
       final int finalCurrent = currentLogId;
       if (!success) {
+
+        if (tserver.getLock() == null || !tserver.getLock().verifyLockAtSource()) {
```

Comment:
There could be a lot of threads running this code for writes in lots of tserers.  If there is temporary HDFS across the closter hiccup it could cause all of those threads to jump on zookeeper.  That is probably ok, so not sure we should try to do anything about that.  Might be worthwhile adding some debug or greater logging when this check is run so that its known that it is happening.

Reply:
Logging added in 77492bf